### PR TITLE
Add Table of Contents to summary_report.html

### LINKED ISSUE #93: Summary report should have a TOC with links
For instance, an import with many SVs like [this one](https://storage.cloud.google.com/unresolved_mcf/fbi/hate_crime/hate_crime_state_city/output/summary_report.html) is hard to navigate without that.

### PR DESCRIPTION
**Pull request by enjoythecode**
Closes #93 

Added a Table of Contents to `summary_report.html` through modifications to its `SummaryReport.ftl` template.

- Links to individual `<h2>` tags
- Sub-links to each counter key, statVar in the first (statVars) table, and place geoID in the second (sample places) table
- Sub-sub-links to each statVar under each geoID in the second (sample places) tables
- Added a `#` character next to each anchor that is linked to from TOC.
  - These characters links to themselves, allowing one to click it and obtain a link to that specific location in the report. This makes it easy to call out specific issues in reviews, as well as allowing convenient linking from written documents.
- Hid the verbose sub- and sub-sub- links under `<details>` tags that are collapsed by default
- Modified the existing styles to keep the TOC with a low profile
- Regenerated goldens since this is a markup modification.

### CODE REVIEW

**Comment by enjoythecode** [comment]
Link to rendered example `summary_report.html` from this branch to preview TOC changes:
https://htmlpreview.github.io/?https://github.com/enjoythecode/import/blob/toc/tool/src/test/resources/org/datacommons/tool/genmcf/statchecks/output/summary_report.html

**Comment by chejennifer** [comment]
Nice change! Instead of the "#" character next to each anchor, can we just make the anchors themselves clickable? The "#" is a little confusing

**Review by chejennifer — COMMENTED** [review]
Nice change! Instead of the "#" character next to each anchor, can we just make the anchors themselves clickable? The "#" is a little confusing

File: util/src/main/resources/SummaryReport.ftl [hunk 1]
```diff
@@ -55,33 +132,40 @@
...
+            </tbody>
+          </#list>
+      </table>
     </div>
     <#if svSummaryMap?has_content>
+      
       <div>
-        <h2>StatVarObservations by StatVar</h2>
+        <h2>
+          <a name="statvars" hrew="#statvars">StatVarObservations by StatVar</a>
```
Comment (chejennifer) [inline util/src/main/resources/SummaryReport.ftl:167]: href

**Review by chejennifer — COMMENTED** [review]
Nice changes, just a few small things

**Review by chejennifer — APPROVED** [review]
thanks for the updates!

**Pull request merged**